src/core/execute.c

```diff
@@ -132,8 +132,7 @@ int exec_context_apply_tty_size(
         /* If we got nothing so far and we are talking to a physical device, and the TTY reset logic is on,
          * then let's query dimensions from the ANSI driver. */
         if (rows == UINT_MAX && cols == UINT_MAX &&
-            context->tty_reset &&
-            terminal_is_pty_fd(output_fd) == 0 &&
+            exec_context_shall_ansi_seq_reset(context) &&
```

Comment:
Maybe, better to add a short comment why this is reused here, as the function named with 'reset', but the function `terminal_get_size_by_dsr()` is not directly related to resetting.